GameConfiguration.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+
+#include <stdint.h>
+
+#include "common.h"
+#include "commonTypes.h"
+
+#include "allegro5/color.h"
+
+struct GameConfiguration {
+    bool overlay_mode;
```

Comment:
why are some of these fields left uninitialized?

Reply:
I propagating the ones that were being explicitly initialized elsewhere to the definition. The remainder are default-initialized (that is, zero-initialized).

The main semantic difference is that before, these values were initialized only on the _first_ load of stonesense; now, they'r reset to their defaults (whether explicitly set here or defaulted) when the new copy of `GameConfiguration` is copied over the global variable in `stonesense_thread` in `main.cpp`.

There is more work to be done here: `GameConfiguration` includes several members (e.g. `ALLEGRO_COND* readCond`) which are clearly _not_ configuration and which should probably be moved to `GameState`. One thing at a time.